core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -235,6 +235,23 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
           }
 
           tryBuilder.map(_.result())
+        case o: ConfigObject =>
```

Comment:
I think we could eliminate some duplicated code and improve the error message when the index is non-numeric if we replaced the entire `from` method with something like this:

```
    // Akin to scalaz's `traverseU` but limited to operating on Try instead of all monads.
    private def traverseTry[A, B, C](i: TraversableOnce[A], builder: Builder[B, C])(f: A => Try[B]): Try[C] = {
      i.foldLeft(Try(builder)) {
        case (builderMaybe, a) => builderMaybe.flatMap { builder =>
          f(a).map(builder += _)
        }
      }.map(_.result)
    }

    override def from(config: ConfigValue): Try[F[T]] = {
      config match {
        case co: ConfigList =>
          traverseTry(co.asScala, cbf())(configConvert.value.from)
        case o: ConfigObject =>
          val objectAsMap: collection.mutable.Map[String, ConfigValue] = o.asScala
          for {
            indexedEntries <- traverseTry(objectAsMap, new collection.mutable.ListBuffer[(Int, ConfigValue)]) {
              case (keyString, value) =>
                Try(keyString.toInt -> value)
                  .recoverWith {
                    case t =>
                      val message = s"Cannot interpet $keyString as a numeric index. Tried to read the object as an indexed sequence. " +
                        "Expecting syntax like (a.0=0, a.1=1, ...) when loading a sequence called 'a'"
                      Failure[Nothing](new IllegalArgumentException(message))
                  }
            }
            sortedValues = indexedEntries.sortBy(_._1).map(_._2)
            result <- traverseTry(sortedValues, cbf())(configConvert.value.from)
          } yield result
        case other =>
          Failure(WrongTypeException(other.valueType().toString))
      }
    }
```

@melrief What do you think about that? Do you have a better name than `traverseTry`?